### PR TITLE
Use function calling to parse responses

### PR DESCRIPTION
**Pull request by ankrgyl**
A substantial fraction (20%+) of responses have parse errors. This change uses function calling to reduce that.

### CODE REVIEW

**Comment by ankrgyl** [comment]
CC @wong-codaio

File: js/llm.ts [hunk 1]
```diff
@@ -2,13 +2,19 @@ import * as yaml from "js-yaml";
...
 import { ChatCache, cachedChatCompletion } from "./oai.js";
 import { templates } from "./templates.js";
 
-const NO_COT_SUFFIX = `Answer the question by printing only a single choice from {{__choices}} (without quotes or punctuation) corresponding to the correct answer with no other text.`;
+const NO_COT_SUFFIX =
+  "Answer the question by calling the `select_choice` function with a single choice from {{__choices}}.";
 
-const COT_SUFFIX = `Write out in a step by step manner your reasoning to be sure that your conclusion is correct. Avoid simply stating the correct answer at the outset. Then print only a single choice from {{__choices}} (without quotes or punctuation) on its own line corresponding to the correct answer. At the end, repeat just the answer by itself on a new line formatted as "Answer=X"`;
+const COT_SUFFIX =
+  "Answer the question by calling the `select_choice` with your reasoning in a step-by-step matter to be sure that your conclusion is correct. Avoid simply stating the correct answer at the outset. Select a single choice by setting the `choice` parameter to a single choice from {{__choices}}.";
```
Comment (manugoyal) [inline js/llm.ts:17]: the `select_choice` function?

File: py/autoevals/llm.py [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
 """.strip().replace(
     "\n", " "
 )
 
 COT_SUFFIX = """\
-Write out in a step by step manner your reasoning to be sure that your conclusion is correct. Avoid
-simply stating the correct answer at the outset. Then print only a single choice from {{__choices}} (without
-quotes or punctuation) on its own line corresponding to the correct answer. At the end, repeat just the
-answer by itself on a new line formatted as "Answer=X"
+Answer the question by calling the `select_choice` with your reasoning in a step-by-step matter to be
```
Comment (manugoyal) [inline py/autoevals/llm.py:25]: same here

**Review by manugoyal — APPROVED** [review]

File: js/llm.ts (hunk 1 above)
Comment (ankrgyl) [reply js/llm.ts:17]: good catch

**Pull request merged**